app/models/intervention.rb

```diff
@@ -227,6 +228,20 @@ def remove_short_links
     short_links.destroy_all
   end
 
+  def cascade_access_type_change
+    return unless shared_to == 'anyone'
+
+    permitted_schedules = %w[after_fill immediately]
+
+    sessions.each do |session|
+      session.update!(
+        schedule: session.schedule.in?(permitted_schedules) ? session.schedule : 'after_fill',
```

Comment:
what do you think to skip if the type of scheduling is correct?

Reply:
Yeah, good idea 😅 Thank you